attributes/default.rb

```diff
@@ -1,5 +1,80 @@
+default['twemproxy']['install_method'] = value_for_platform_family(
+  'rhel' => 'source',
+  'debian' => 'package'
+)
+
+# twemproxy user / group, set to root if ubuntu package install
+default['twemproxy']['user'] = node['twemproxy']['install_method'] == 'package' ? 'root' : 'twemproxy'
+default['twemproxy']['group'] = node['twemproxy']['install_method'] == 'package' ? 'root' : 'twemproxy'
+# setup user, used by source install
+default['twemproxy']['manage_user'] = true
+default['twemproxy']['home'] = nil
+
+default['twemproxy']['config_dir'] = '/etc/twemproxy'
+default['twemproxy']['config_file'] = ::File.join(node['twemproxy']['config_dir'], 'config.yml')
+
+default['twemproxy']['umask']         = '0022'
+default['twemproxy']['dir_mode']      = '0755'
+default['twemproxy']['parent_dir']    = '/usr/local/twemproxy'
+default['twemproxy']['log_dir']       = '/var/log/twemproxy'
+default['twemproxy']['log_file']       = ::File.join(node['twemproxy']['log_dir'], 'twemproxy.log')
+default['twemproxy']['conf_dir']      = '/etc/twemproxy'
+default['twemproxy']['data_dir']      = '/opt/twemproxy'
+
+# for tarball installation
+default['twemproxy']['version_dir']   = ::File.join(node['twemproxy']['parent_dir'], node['twemproxy']['version'])
+default['twemproxy']['install_dir']   = ::File.join(node['twemproxy']['version_dir'], 'installed')
+default['twemproxy']['source_dir']    = ::File.join(node['twemproxy']['version_dir'], "twemproxy-#{node['twemproxy']['version']}")
+
+default['twemproxy']['notify_restart']  = true # notify service restart on config change
+default['twemproxy']['service_name']    = 'twemproxy'
+
+# twemproxy daemon args
+default['twemproxy']['port']          = 8_983
+default['twemproxy']['verbosity']     = 5
+default['twemproxy']['stats_port']    = 22_222
+default['twemproxy']['stats_address'] = node['ipaddress']
+default['twemproxy']['stats_interval'] = 30_000
+default['twemproxy']['mbuf_size'] = 16_384
+default['twemproxy']['pid_file'] = nil
+
+default['twemproxy']['config'] = {}
+
+default['twemproxy']['force_compile'] = false
+default['twemproxy']['compile']['--prefix'] = node['twemproxy']['install_dir']
+default['twemproxy']['compile']['--sysconfdir'] = node['twemproxy']['conf_dir']
+default['twemproxy']['compile']['--datadir']    = node['twemproxy']['data_dir']
+
+# twemproxy process limits
+default['twemproxy']['limits']['memlock'] = 'unlimited'
+default['twemproxy']['limits']['nofile']  = 48_000
+default['twemproxy']['limits']['nproc']   = 'unlimited'
+
+default['twemproxy']['daemon_prefix'] = case node['twemproxy']['install_method']
+                                        when 'source'
+                                          ::File.join(node['twemproxy']['install_dir'], 'bin')
+                                        else
+                                          '/usr/bin'
+                                        end
+
+default['twemproxy']['service_config_file'] = value_for_platform_family(
```

Comment:
@guilhem this may be overwhelming for users who does not have systemd included in base OS (e.g. amazon). i think we can make it optional, something like configurable via node attribute `node['twemproxy']['service_provider']`, what do you think about it?